packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -205,9 +205,10 @@ export class AtomicSearchBox {
           part="suggestion"
           id={id}
           class="suggestion h-9 px-2 cursor-pointer text-left text-sm bg-transparent border-none shadow-none flex flex-row items-center"
-          innerHTML={suggestion.highlightedValue}
           value={index}
-        />
+        >
+          <pre class="font-sans" innerHTML={suggestion.highlightedValue}></pre>
+        </li>
```

Comment:
Could `highlightedValue` become a recurring issue with headless, in that value needs to be rendered inside this specific html tag?

Considering `highlightedValue` is html, intended for the web, maybe its okay to replace spaces with nbsp as you suggested. The property is already locked-in to the run-time environment.

If not, then we should at least add to the doc annotation that this attribute should be rendered inside a `pre`. My preference would be to have things just work OOTB though,

Reply:
Seconded

Reply:
If that's a recurring theme (betting it will be), then we could look into adding a dedicated "highlighting for HTML" function in headless.

Reply:
True for the delimiters, but we are escaping the suggestion before adding the delimiters. So there is an implicit string transformation to defend against a vulnerability because we are assuming that the string will be rendered as html.

Is transforming spaces different?

Ref: https://github.com/coveo/ui-kit/blob/master/packages/headless/src/utils/highlight.ts#L111